src/modules/auth/auth.ts

```diff
@@ -163,6 +170,36 @@ export class Auth {
     }
   }
 
+  async initiateClientFlow(tokenGetParam?: any) {
+    if (!this.#onAuthSuccess) {
+      throw new Error(
+        "'onAuthSuccess' was not provided. Please provide the 'onAuthSuccess' function"
+      );
+    }
+    if (!this.#clientId || !this.#clientSecret) {
+      throw new Error(
+        "Cannot get access token because 'clientId' or 'clientSecret' is missing"
+      );
+    }
+    const payload: ClientTokenRequestPayload = {
+      client_id: this.#clientId!,
+      client_secret: this.#clientSecret!,
+      grant_type: GrantType.CLIENT_CREDENTIALS
+    };
+    try {
+      const result = await this.#post("/token", payload);
+      const { access_token: accessToken } = result;
+      return this.#onAuthSuccess(accessToken, tokenGetParam);
+    } catch (err: any) {
+      const error = err instanceof Error ? err : new Error('Failed to get access token');
+      const errorMessage = `Failed to get access token: ${error.message}`;
```

Comment:
In case of `err` not being an instance of Error, the `errorMessage` will become "Failed to get access token: Failed to get access token". Let's modify this a little bit so that we don't get a duplicated message.

```suggestion
      const errorMessage = err instanceof Error ? `Failed to get access token: ${err.message}` : "Failed to get access token";
```

Reply:
This makes much more sense, thanks for catching.